### PR TITLE
add snapshot publish func

### PR DESCRIPTION
**Pull request by benPearce1**

### CODE REVIEW

File: pkg/runbooks/runbook_snapshot_service.go [hunk 1]
```diff
@@ -43,6 +44,31 @@ func (s *RunbookSnapshotService) Add(runbookSnapshot *RunbookSnapshot) (*Runbook
...
+	if err := runbookSnapshot.Validate(); err != nil {
+		return nil, internal.CreateValidationFailureError(constants.OperationAdd, err)
+	}
+
+	path, err := services.GetAddPath(s, runbookSnapshot)
+	if err != nil {
+		return nil, err
+	}
+
+	path = fmt.Sprintf("%s?publish=true", path)
```
Comment (benPearce1) [inline pkg/runbooks/runbook_snapshot_service.go:62]: there is not "command" to publish a snapshot, this parameter is only available as a query parameter and setting it to any value causes it to publish https://github.com/OctopusDeploy/OctopusDeploy/blob/main/source/Octopus.Core/Features/RunbookSnapshots/CreateRunbookSnapshotCommandHandler.cs#L92

File: pkg/runbooks/runbook_snapshot_service.go [hunk 2]
```diff
@@ -35,6 +45,10 @@ func (s *RunbookSnapshotService) Add(runbookSnapshot *RunbookSnapshot) (*Runbook
 		return nil, err
 	}
 
+	if publish {
+		path = fmt.Sprintf("%s?publish=true", path)
```
Comment (benPearce1) [inline pkg/runbooks/runbook_snapshot_service.go:49]: there is not "command" to publish a snapshot, this parameter is only available as a query parameter and setting it to any value causes it to publish https://github.com/OctopusDeploy/OctopusDeploy/blob/main/source/Octopus.Core/Features/RunbookSnapshots/CreateRunbookSnapshotCommandHandler.cs#L92

Comment (IsaacCalligeros95) [reply pkg/runbooks/runbook_snapshot_service.go:49]: Reading go always confuses me... Is it possible to have previous query parameters here? If so do we need to check the existing path?

Comment (benPearce1) [reply pkg/runbooks/runbook_snapshot_service.go:49]: yep, spot on. The default path has no parameters on it.

**Review by IsaacCalligeros95 — APPROVED** [review]
LGTM

File: pkg/runbooks/runbook_snapshot_service.go (hunk 2 above)
Comment (benPearce1) [reply pkg/runbooks/runbook_snapshot_service.go:49]: I am not going to fix this now. It will probably never become a problem

**Pull request merged**